nifi-nar-bundles/nifi-asana-bundle/nifi-asana-processors/src/main/java/org/apache/nifi/processors/asana/GetAsanaObject.java

```diff
@@ -0,0 +1,430 @@
+            }
+        }
+
+        if (newItems.isEmpty() && updatedItems.isEmpty() && removedItems.isEmpty()) {
+            context.yield();
+            getLogger().debug("Yielding, as there are no new FlowFiles.");
+            return;
+        }
+
+        session.transfer(newItems, REL_NEW);
+        session.transfer(updatedItems, REL_UPDATED);
+        session.transfer(removedItems, REL_REMOVED);
+
+        session.commitAsync();
+        Map<String, String> state = objectFetcher.saveState();
+        try {
+            persistState(state, context);
+        } catch (IOException e) {
+            throw new ProcessException(e);
+        }
+        getLogger().debug(
+            "New state after transferring {} new, {} updated, and {} removed items: {}",
+            newItems.size(), updatedItems.size(), removedItems.size(), state);
+    }
+
+    protected AsanaObjectFetcher createObjectFetcher(final ProcessContext context, AsanaClient client) {
+        final String objectType = context.getProperty(PROP_ASANA_OBJECT_TYPE).getValue();
+        final String projectName = context.getProperty(PROP_ASANA_PROJECT).getValue();
+        final String sectionName = context.getProperty(PROP_ASANA_SECTION).getValue();
+        final String teamName = context.getProperty(PROP_ASANA_TEAM_NAME).getValue();
+        final String tagName = context.getProperty(PROP_ASANA_TAG).getValue();
+
+        switch (objectType) {
+            case AV_NAME_COLLECT_TASKS:
+                return new AsanaTaskFetcher(client, projectName, sectionName, tagName);
+            case AV_NAME_COLLECT_PROJECTS:
+                return new AsanaProjectFetcher(client);
+            case AV_NAME_COLLECT_PROJECT_EVENTS:
+                return new AsanaProjectEventFetcher(client, projectName);
+            case AV_NAME_COLLECT_PROJECT_MEMBERS:
+                return new AsanaProjectMembershipFetcher(client, projectName);
+            case AV_NAME_COLLECT_PROJECT_STATUS_ATTACHMENTS:
+                return new AsanaProjectStatusAttachmentFetcher(client, projectName);
+            case AV_NAME_COLLECT_PROJECT_STATUS_UPDATES:
+                return new AsanaProjectStatusFetcher(client, projectName);
+            case AV_NAME_COLLECT_STORIES:
+                return new AsanaStoryFetcher(client, projectName, sectionName, tagName);
+            case AV_NAME_COLLECT_TAGS:
+                return new AsanaTagFetcher(client);
+            case AV_NAME_COLLECT_TASK_ATTACHMENTS:
+                return new AsanaTaskAttachmentFetcher(client, projectName, sectionName, tagName);
+            case AV_NAME_COLLECT_TEAMS:
+                return new AsanaTeamFetcher(client);
+            case AV_NAME_COLLECT_TEAM_MEMBERS:
+                return new AsanaTeamMemberFetcher(client, teamName);
+            case AV_NAME_COLLECT_USERS:
+                return new AsanaUserFetcher(client);
+        }
+
+        throw new RuntimeException("Cannot fetch objects of type: " + objectType);
```

Comment:
Please do not throw raw `RuntimeException`s, use NiFi's `ProcessException` instead.

Reply:
Done in [dc2a050](https://github.com/apache/nifi/pull/6504/commits/dc2a050548a389d89181670371748af741eefd61)